policy/policy_test.go

```diff
@@ -131,3 +135,231 @@ func TestPolicyRead(t *testing.T) {
+}
+
+func TestParseFromFile(t *testing.T) {
+	t.Parallel()
+
+	// Provide the path to the policy file
+	policyFile := "testdata/policy-ok.yaml"
+
+	// Call the function being tested
+	sp, err := ParseFromFile(policyFile)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if len(sp.Policies) != 3 {
+		t.Errorf("Unexpected number of policies: got %v, want %v", len(sp.Policies), 3)
+	}
+	invalidPolicy := "testdata/policy-invalid-score-0.yaml"
+	_, err = ParseFromFile(invalidPolicy)
+	if err == nil {
+		t.Error("Expected an error")
+	}
+	invalidMode := "testdata/policy-invalid-mode.yaml"
+	_, err = ParseFromFile(invalidMode)
+	if err == nil {
+		t.Error("Expected an error")
+	}
+	invalidFile := "testdata/invalid-file.yaml"
+	_, err = ParseFromFile(invalidFile)
+	if err == nil {
+		t.Error("Expected an error")
+	}
+}
+
+func TestModeToProto(t *testing.T) {
+	t.Parallel()
+
+	// Call the function being tested
+	mode := modeToProto("enforced")
+
+	// Check the result
+	expectedMode := CheckPolicy_ENFORCED
+	if mode != expectedMode {
+		t.Errorf("Unexpected mode. Got: %v, Want: %v", mode, expectedMode)
+	}
+
+	// Call the function again with a different mode
+	mode = modeToProto("disabled")
+
+	// Check the result
+	expectedMode = CheckPolicy_DISABLED
+	if mode != expectedMode {
+		t.Errorf("Unexpected mode. Got: %v, Want: %v", mode, expectedMode)
+	}
+
+	// Test panic with an unknown mode
+	testPanic := func() {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Errorf("Expected panic, but no panic occurred")
+			}
+		}()
+		modeToProto("unknown")
+	}
+
+	// Run the panic test
+	testPanic()
+
+	// Additional test cases can be added to cover more scenarios
+}
+
+func TestGetEnabled(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                  string
+		sp                    *ScorecardPolicy
+		argsChecks            []string
+		requiredRequestTypes  []checker.RequestType
+		expectedEnabledChecks int
+		expectedError         bool
+	}{
+		{
+			name:                  "With ScorecardPolicy and argsChecks",
+			sp:                    &ScorecardPolicy{},
+			argsChecks:            []string{"Binary-Artifacts"},
+			requiredRequestTypes:  []checker.RequestType{checker.FileBased, checker.CommitBased},
+			expectedEnabledChecks: 0,
+			expectedError:         true,
+		},
+		{
+			name:                  "With ScorecardPolicy and unsupported check",
+			sp:                    &ScorecardPolicy{},
+			argsChecks:            []string{"Binary-Artifacts", "UnsupportedCheck"},
+			requiredRequestTypes:  []checker.RequestType{checker.FileBased, checker.CommitBased},
+			expectedEnabledChecks: 0,
+			expectedError:         true,
+		},
+		{
+			name:                  "Without ScorecardPolicy and argsChecks",
+			sp:                    nil,
+			argsChecks:            []string{},
+			requiredRequestTypes:  []checker.RequestType{checker.FileBased, checker.CommitBased},
+			expectedEnabledChecks: 4,
```

Comment:
is it worth clarifying where this number comes from in a comment?

Reply:
I will do that.